### DISCUSSION TITLE
gatsby-transformer-csv sort by integer field not working, is sorting as string

### DISCUSSION THREAD

**Discussion by AdamQuadmon** [gatsbyjs/gatsby#26607] [Help]
## Summary
I'm using gatsby-transformer-csv.
I want to sort by fieldNumber as integer.
Values are sorted as string even if displayed as integer using `createTypes` in `gatsby-node.js`
There is also a SO question from another user:
https://stackoverflow.com/questions/63508542/how-can-gatsby-graphql-sort-an-integer-field-numerically-rather-than-alphabetica

## Relevant information

gatsby-node.js
```
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions
  const typeDefs = `
    type TestCsv implements Node @infer {
      numberField: Int
    }
  `
  createTypes(typeDefs)
}
```

test.csv 
```
numberField
1
2
10
11
20
```

query
```
query MyQuery {
  allTestCsv(sort: {fields: numberField, order: DESC}) {
    nodes {
      numberField
    }
  }
}
```

output
```
{
  "data": {
    "allTestCsv": {
      "edges": [
        {
          "node": {
            "numberField": 1
          }
        },
        {
          "node": {
            "numberField": 10
          }
        },
        {
          "node": {
            "numberField": 11
          }
        },
        {
          "node": {
            "numberField": 2
          }
        },
        {
          "node": {
            "numberField": 20
          }
        }
      ]
    }
  },
  "extensions": {}
}
```
If I don't use the createType, values are displayed as strings and the order is the same
```
{
  "data": {
    "allTestCsv": {
      "edges": [
        {
          "node": {
            "numero": "1"
          }
        },
        {
          "node": {
            "numero": "10"
          }
        },
        {
          "node": {
            "numero": "11"
          }
        },
        {
          "node": {
            "numero": "2"
          }
        },
        {
          "node": {
            "numero": "20"
          }
        }
      ]
    }
  },
  "extensions": {}
}
```

**Comment by blainekasten** [comment]
I think @pvdz might know the answer to this. He was working on some performance related work in those files. or @vladar might know too

**Comment by samajammin** [comment]
@AdamQuadmon were you able to figure this out? I'm having the same issue. Thanks in advance!

**Comment by pvdz** [comment]
The Gatsby plugin is a very thin layer on top of the `csvtojson` package.

Additionally, according to the readme, it should generate numbers. These should be sortable numerically https://www.gatsbyjs.com/plugins/gatsby-transformer-csv/#parsing-algorithm

So I guess we'd need tos ee an example of your input csv to figure out what's up :) I don't know about Adam's issue (and I'm sorry I missed the ping).
